### ISSUE TITLE
Identifying Non-Primitive Paramaterized Inputs in Adapter vs Console

### ISSUE THREAD

**Issue by BgRva** [nunit/nunit3-vs-adapter#34]
This will probably be a low-priority issue but adding it anyway.  I have a paramaterized test which uses a TestCaseSourceAttribute for which the input is the IEnumerator for a non-primitive type called "DummyItem".  The purpose of the DummyItem class is solely to test non-primitive inputs to generic methods.  

```csharp
public class DummyItem
{
    public DummyItem(int id, string name)
    {
        Id = id;
        Name = name;
    }
    public int Id { get; set; }
    public string Name { get; set; }
}    

public class DataOfItem : IEnumerable
{
    public IEnumerator GetEnumerator()
    {
        yield return new DummyItem(0, "");
        yield return new DummyItem(3, "BLAH");
    }
}
```

I apply this to the following test method:

```csharp
[TestCaseSource(typeof(DataOfItem))]
public void Creates_And_Adds_To_Graph_Node_Of_Item(DummyItem uniqueId)
{
    //Arrange
    var graph = new Graph<DummyItem>();

    //Act
    var node = graph.AddNode(uniqueId);

    //Assert
    Assert.NotNull(node);
    Assert.AreEqual(uniqueId, node.UniqueId);
    Assert.AreEqual(1, graph.NodeCount);
    Assert.AreEqual(0, graph.EdgeCount);
}
```

The NUnit Adapter only picks up one of the tests for this method and after test discovery displays the following:

```
A test with the same name 'x.Graph.Core.Tests.Default.GraphFixture+AddNode.Creates_And_Adds_To_Graph_Node_Of_Item(x.Graph.Core.Tests.Support.DummyItem)' already exists. This test is not added to the test window.
```
     
But the console runner does pick them up as separate tests.  

If I add an override for the ToString() method to the DummyItem class, the NUnit adapter will identify the individual tests so long as the output of ToString() is unique for each instance of the DummyItem objects returned in the IEnumerator.  At the very least, overriding the ToString() method may be a preferable practice because it allows for the identification of each individual test in the console runner and VS adapter. 

![console_output_complex_types](https://cloud.githubusercontent.com/assets/3044102/7788259/60561fcc-0201-11e5-9885-92ebdb086e7f.PNG)

I would be willing to update some documentation regarding this if it would help.

**Comment by OsirisTerje** [comment]
The message is misleading.  The Test Explorer picks up both, but it displays them as one test with two results. 
![image](https://cloud.githubusercontent.com/assets/203432/7789379/c5c0d5ac-025e-11e5-81c0-abf1958977a2.png)

The documentation should surely be updated on this kind of tests, and probably the  message in the Output window should be made better.

**Comment by CharliePoole** [comment]
If we can tell that the apparent error is caused by two tests using the same method with different arguments, I think we could suppress the warning. However, I'm not sure we can distinguish this situation from true duplicates.

Maybe this should be labeled as a documentation problem and we should start a new issue (idea?) about changing how we handle parametrized test cases. The MS team had suggested to me at one point that we treat _all_ the cases as separate results of a single test.

I'm not sure this is the right thing to do, which is why I would call it merely an idea right now. What if somebody has thousands of cases? How will that look in the adapter?

**Comment by CharliePoole** [comment]
@OsirisTerje This has slipped through the cracks I guess. You marked it as a low-priority easy fix in May and I asked you some questions in a comment but didn't follow up to remind you.

Basically, I don't see how this is an easy fix. In my comment above I explain why. Could you explain what you meant?

**Comment by OsirisTerje** [comment]
What I meant was that the message is misleading.  It comes from vstest and not from us.  And, the tests are being picked up, so all we can do is to add our own message to this.  
We can do a test for thousands of cases of course, but it is the Test Explorer that must handle that. 
I set it as low pri because it is no real error.

**Comment by CharliePoole** [comment]
I see what you mean - it's easy if we don't try to distinguish the different cases and don't use the approach of redefining what we report as a test to VS, but  instead just document it. I categorized it as a documentation issue. I had not designated it previously as either a bug or a documentation issue, because we were still discussing it. I think this project will be in your hands before we would get to any deeper fix, so it's your call.

Because 'easyfix' is used to steer new contributors to code they can work on, it's not useful for this one, so I removed it.
